### PR TITLE
saml idp with okta updates

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/docs/issues/4086

Updated:
- Self-Managed Admin Guide > Onboard Users > SAML Single Sign-On (E20) > Set Up a Connection App for Mattermost Single Sign-On
    - Updated Step #9 to include the ID attribute mapped to ``user.id`` with an updated screenshot
- Self-Managed Admin Guide > Onboard Users > SAML Single Sign-On (E20) > Configure SAML Sign-in for Mattermost
   - Updated Step #5 to include the ID attribute mapped to ``user.id`` with an updated screenshot

### CODE REVIEW

**Comment by cwarnermm** [comment]
@icelander - Where would be the best place in this product documentation to mention that when using Okta as LDAP, customers need to use ``uniqueIdentifier`` as the Mattermost User ID attribute?

File: source/deployment/sso-saml-okta.rst [hunk 1]
```diff
@@ -44,30 +42,38 @@ If you’d like to use a Mattermost logo for the application, you can download o
 
 	.. image:: ../../source/images/okta_5_advanced_saml_settings.PNG
 
-9. Enter attribute statements, which will be used to map attributes between Okta and Mattermost. For more information on which attributes are configurable, see our :ref:`documentation on SAML configuration settings <saml-enterprise>`. Email and username attributes are required. For Mattermost servers running 3.3 and earlier, first name and last name attributes are also required.
+9. Enter attribute statements used to map attributes between Okta and Mattermost. For more information on which attributes are configurable, see our `documentation on SAML configuration settings <https://docs.mattermost.com/administration/config-settings.html#saml>`__. Email and username attributes are required. For SAML with Okta, an `ID attribute <https://docs.mattermost.com/administration/config-settings.html#id17>`__ is also required, and that ID must be mapped to ``user.id``. For Mattermost servers running 3.3 and earlier, first name and last name attributes are also required.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-okta.rst:45]: How many servers are still using 3.3? Maybe we can take this line out? :)

File: source/deployment/sso-saml-okta.rst [hunk 2]
```diff
@@ -44,30 +42,38 @@ If you’d like to use a Mattermost logo for the application, you can download o
...
+
+16. Download the X.509 Certificate file and save it. You may need to upload it to Mattermost in a later step.
 
 	.. image:: ../../source/images/okta_9_view_instructions.PNG
 
 Configure SAML Sign-in for Mattermost
 --------------------------------------
 
-Start the Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML**, and paste the copied Identity Provider Metadata URL in the **Identity Provider Metadata URL** field, then select **Get SAML Metadata from IdP**.
+Start the Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML 2.0**, then paste the copied Identity Provider Metadata URL in the **Identity Provider Metadata URL** field, and select **Get SAML Metadata from IdP**.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-okta.rst:74]:
```suggestion
Start the Mattermost server and sign into Mattermost as a System Admin. Go to **System Console > Authentication > SAML 2.0**, then paste the copied Identity Provider Metadata URL in the **Identity Provider Metadata URL** field, and select **Get SAML Metadata from IdP**.
```

File: source/deployment/sso-saml-okta.rst [hunk 3]
```diff
@@ -88,11 +94,11 @@ Alternatively you can enter the following fields manually:
...
 4. Configure Mattermost to sign SAML requests using the Service Provider Private Key.
 
-5. Set attributes for the SAML Assertions, which will be used to update user information in Mattermost. Attributes for email and username are required and should match the values you entered in Okta earlier. See :ref:`documentation on SAML configuration settings <saml-enterprise>` for more detail.
-
-For Mattermost servers running 3.3 and earlier, the first name and last name attributes are also required fields.
-
-	.. image:: ../../source/images/okta_13_mattermost_attributes.PNG
+5. Set attributes for the SAML Assertions used to update user information in Mattermost. 
+	- Attributes for Email, Username, and Id are required and should match the values you entered in Okta earlier.
+	- For Mattermost servers running 3.3 and earlier, the first name and last name attributes are also required fields.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-okta.rst:99]: See comment above. :)

File: source/deployment/sso-saml-okta.rst [hunk 4]
```diff
@@ -44,30 +42,38 @@ If you’d like to use a Mattermost logo for the application, you can download o
 
 	.. image:: ../../source/images/okta_5_advanced_saml_settings.PNG
 
-9. Enter attribute statements, which will be used to map attributes between Okta and Mattermost. For more information on which attributes are configurable, see our :ref:`documentation on SAML configuration settings <saml-enterprise>`. Email and username attributes are required. For Mattermost servers running 3.3 and earlier, first name and last name attributes are also required.
+9. Enter attribute statements used to map attributes between Okta and Mattermost. For more information on which attributes are configurable, see our `documentation on SAML configuration settings <https://docs.mattermost.com/administration/config-settings.html#saml>`__. Email and username attributes are required. For SAML with Okta, an `ID attribute <https://docs.mattermost.com/administration/config-settings.html#id17>`__ is also required, and that ID must be mapped to ``user.id``. For Mattermost servers running 3.3 and earlier, first name and last name attributes are also required.
 
-	.. image:: ../../source/images/okta_6_attribute_statements.PNG
+	.. image:: ../../source/images/okta_6_attribute_statements.png
```
Comment (justinegeffen) [inline source/deployment/sso-saml-okta.rst:47]: Need to confirm that the extension is lowercase as it doesn't look like it's showing in preview now.

File: source/deployment/sso-saml-okta.rst [hunk 5]
```diff
@@ -44,30 +42,38 @@ If you’d like to use a Mattermost logo for the application, you can download o
...
 
 	.. image:: ../../source/images/okta_9_view_instructions.PNG
 
 Configure SAML Sign-in for Mattermost
 --------------------------------------
 
-Start the Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML**, and paste the copied Identity Provider Metadata URL in the **Identity Provider Metadata URL** field, then select **Get SAML Metadata from IdP**.
+Start the Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML 2.0**, then paste the copied Identity Provider Metadata URL in the **Identity Provider Metadata URL** field, and select **Get SAML Metadata from IdP**.
+
+        .. image:: ../../source/images/saml-identity-provider-metadata-url.PNG
```
Comment (justinegeffen) [inline source/deployment/sso-saml-okta.rst:76]: Has this image been uploaded? It's not showing in preview but that's probably not an issue if it's a new image and part of this PR.

File: source/deployment/sso-saml-okta.rst [hunk 6]
```diff
@@ -88,11 +94,11 @@ Alternatively you can enter the following fields manually:
...
-5. Set attributes for the SAML Assertions, which will be used to update user information in Mattermost. Attributes for email and username are required and should match the values you entered in Okta earlier. See :ref:`documentation on SAML configuration settings <saml-enterprise>` for more detail.
-
-For Mattermost servers running 3.3 and earlier, the first name and last name attributes are also required fields.
-
-	.. image:: ../../source/images/okta_13_mattermost_attributes.PNG
+5. Set attributes for the SAML Assertions used to update user information in Mattermost. 
+	- Attributes for Email, Username, and Id are required and should match the values you entered in Okta earlier.
+	- For Mattermost servers running 3.3 and earlier, the first name and last name attributes are also required fields.
+	
+	.. image:: ../../source/images/okta_13_mattermost_attributes.png
```
Comment (justinegeffen) [inline source/deployment/sso-saml-okta.rst:101]: This image isn't displaying in preview.

File: source/deployment/sso-saml-okta.rst (hunk 1 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:45]: Given that v3.3 servers aren't represented in our [Looker-based Server & Client Version Usage data](https://mattermost.looker.com/dashboards/172?Mobile%20Client%20Usage=&filter_config=%7B%22Mobile%20Client%20Usage%22:%5B%7B%22type%22:%22any%22,%22values%22:%5B%7B%22constant%22:%22%22%7D,%7B%7D%5D,%22id%22:1%7D%5D%7D), and that our data only starts at v4.0, under most circumstances, I feel we'd be safe to remove this information. However, I'd like to complete some due diligence first to ensure we won't be negatively impacting customers relying on these details by removing them.

File: source/deployment/sso-saml-okta.rst (hunk 5 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:76]: This appears to have been a pre-existing image file that's missing. It's not a new file added as part this PR work.

File: source/deployment/sso-saml-okta.rst (hunk 4 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:47]: There were 2 files with identical file names (except that one extension was PNG and the other was png). I've removed the older file as part of this PR.

File: source/deployment/sso-saml-okta.rst (hunk 6 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:101]: There were 2 files with identical file names (except that one extension was PNG and the other was png). I've removed the older file as part of this PR.

**Comment by cwarnermm** [comment]
@justinegeffen - Could you try building this PR, please? I continue to be unable to see builds locally.

File: source/deployment/sso-saml-okta.rst (hunk 1 above)
Comment (icelander) [reply source/deployment/sso-saml-okta.rst:45]: I'm comfortable with removing 3.3. It's way, way past supported so the first they should do is upgrade

**Review by icelander — APPROVED** [review]
LGTM! I'm fine with removing mention of version 3.3, too

File: source/deployment/sso-saml-okta.rst (hunk 1 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:45]: Thanks, @icelander! I'll remove the references to v3.3 here.

File: source/deployment/sso-saml-okta.rst (hunk 5 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:76]: I've confirmed that this image is also missing on Production.

File: source/deployment/sso-saml-okta.rst (hunk 4 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:47]: Confirmed that the correct screenshot is displaying now.

File: source/deployment/sso-saml-okta.rst (hunk 6 above)
Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:100]: Confirmed that the correct image is displaying now.

File: source/deployment/sso-saml-okta.rst (hunk 5 above)
Comment (amyblais) [reply source/deployment/sso-saml-okta.rst:76]: @justinegeffen / @cwarnermm The image is currently broken on production https://docs.mattermost.com/deployment/sso-saml-okta.html#configure-saml-sign-in-for-mattermost

Comment (cwarnermm) [reply source/deployment/sso-saml-okta.rst:76]: Thanks, @amyblais. Fixing this with https://github.com/mattermost/docs/pull/4530

**Pull request merged**